src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1240,146 +1240,153 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
+                                            if (hasLocalClusterTx == hasRemoteClusterTx)
+                                            {
+                                                // when hasLocalClusterTx and hasRemoteClusterTx both 'true'
+                                                // it is a case of a conflict between documents which were modified in a cluster transaction
+                                                // in two _different clusters_, so we will treat it as a "normal" conflict
+
+                                                IsIncomingReplication = false;
+                                                _replicationInfo.ConflictManager.HandleConflictForDocument(context, doc.Id, doc.Collection, doc.LastModifiedTicks,
+                                                    document, rcvdChangeVector, doc.Flags);
+                                                continue;
+                                            }
 
-                                        // cluster tx has precedence over regular tx
+                                            // cluster tx has precedence over regular tx
 
-                                        if (hasLocalClusterTx)
-                                            goto case ConflictStatus.AlreadyMerged;
+                                            if (hasLocalClusterTx)
+                                                goto case ConflictStatus.AlreadyMerged;
 
-                                        if (hasRemoteClusterTx)
-                                            goto case ConflictStatus.Update;
+                                            if (hasRemoteClusterTx)
+                                                goto case ConflictStatus.Update;
 
-                                        break;
+                                            break;
 
-                                    case ConflictStatus.AlreadyMerged:
-                                        // we have to do nothing here
-                                        break;
+                                        case ConflictStatus.AlreadyMerged:
+                                            // we have to do nothing here
+                                            break;
 
-                                    default:
-                                        throw new ArgumentOutOfRangeException(nameof(conflictStatus),
-                                            "Invalid ConflictStatus: " + conflictStatus);
+                                        default:
+                                            throw new ArgumentOutOfRangeException(nameof(conflictStatus),
+                                                "Invalid ConflictStatus: " + conflictStatus);
+                                    }
+                                }
+                                catch (Exception e)
+                                {
+                                    throw new InvalidOperationException($"Doc id:\"{doc.Id}\"", e);
```

Comment:
```suggestion
                                    throw new InvalidOperationException($"Failed to replicate document: {doc.Id}", e);
```

Also, what about failures for attachments, counters, time series, tombstones, etc?